trieste/acquisition/optimizer.py

```diff
@@ -177,7 +184,8 @@ def _objective() -> TensorType:
 
         def _perform_optimization(starting_point: TensorType, bounds: spo.Bounds) -> OptimizeResult:
             variable.assign(starting_point)  # [1, D]
-            return gpflow.optimizers.Scipy().minimize(_objective, (variable,), bounds=bounds)
+            optimizer_args["bounds"] = bounds
+            return gpflow.optimizers.Scipy().minimize(_objective, (variable,), **optimizer_args)
```

Comment:
Slightly concerned about modifying global (relatively to this function) dictionary. This call leaves the args dict modified after the call, which is a side effect that can be dangerous. Would be safer to create a copy of the dict.

Reply:
thanks for spotting that, resolved!